### ISSUE TITLE
[bug] Moon main thread panicked while using git version 2.43.0.vfs.0.0

### ISSUE THREAD

**Issue by huandzh** [moonrepo/moon#1281]
**Describe the bug**

moon main thread panicked with the installed Git version "2.43.0.vfs.0.0".

```shell
moon run app:build
Error:   × Main thread panicked.
  ├─▶ at nextgen\vcs\src\git.rs:639:36
  ╰─▶ called `Result::unwrap()` on an `Err` value: Error("unexpected character '.' after patch version number")
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Steps to reproduce**

1. has a git installation with version 2.43.0.vfs.0.0
2. run any task from the root of the moon repo

**Expected behavior**

moon can run tasks

**Environment**

```
  System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
    Memory: 3.24 GB / 15.93 GB
  Binaries:
    Node: 21.5.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
  Managers:
    Cargo: 1.70.0 - ~\.cargo\bin\cargo.EXE
    pip3: 23.0.1 - C:\Python311\Scripts\pip3.EXE
  Utilities:
    CMake: 3.23.0
    Git: 2.43.0.
    Clang: 16.0.2 - C:\Program Files\LLVM\bin\clang.EXE
    Curl: 8.0.1 - C:\Windows\system32\curl.EXE
  Virtualization:
    Docker: 24.0.6 - C:\Program Files\Docker\Docker\resources\bin\docker.EXE
  IDEs:
    VSCode: 1.85.1 - C:\Users\p\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
    Visual Studio: 17.8.34408.163 (Visual Studio Community 2022)
  Languages:
    Bash: 5.1.4 - C:\Windows\system32\bash.EXE
    Python: 3.11.0
    Rust: 1.70.0
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (120.0.2210.144), ChromiumDev (122.0.2348.0)
    Internet Explorer: 11.0.19041.1566
```

**Additional context**

```shell
git --version
git version 2.43.0.vfs.0.0
```

The issue is similar to #945 . possible source: https://github.com/huandzh/moon/blob/fb3c05016e6537acf50f7bf2a384017e89b3961f/nextgen/vcs/src/git.rs#L29-L52

**Comment by huandzh** [comment]
btw. 2.43.0.vfs.0.0 is the version number from [VFSForGit](https://github.com/microsoft/vfsforgit).

**Comment by milesj** [comment]
Fixed in 1.20
